tests/metrics/network/iperf3_kubernetes/k8s-network-metrics-iperf3.sh

```diff
@@ -100,79 +99,77 @@ function iperf3_jitter() {
 	export jitter_result=$(printf "%0.3f\n" $result)
 	export jitter_units="ms"
 
-	if [ "$COLLECT_ALL" == "true" ]; then
-		iperf3_all_collect_results
-	else
-		metrics_json_init
-		metrics_json_start_array
-
-		local json="$(cat << EOF
-		{
-			"jitter": {
-				"Result" : $jitter_result,
-				"Units" : "ms"
-			}
+	[ -z "${jitter_result}" ] && die "Iperf3 was unable to collecte Jitter results."
+	[ "$COLLECT_ALL" == "true" ] && return
+
+	metrics_json_init
+	metrics_json_start_array
+
+	local json="$(cat << EOF
+	{
+		"jitter": {
+			"Result" : "${jitter_result}",
+			"Units" : "ms"
 		}
+	}
 EOF
 )"
-		metrics_json_add_array_element "$json"
-		metrics_json_end_array "Results"
-	fi
+	metrics_json_add_array_element "${json}"
+	metrics_json_end_array "Results"
 }
 
 function iperf3_parallel() {
 	# This will measure four parallel connections with iperf3
 	kubectl exec -i "$client_pod_name" -- sh -c "iperf3 -J -c ${server_ip_add} -P 4" | jq '.end.sum_received.bits_per_second' > "${iperf_file}"
-	export parallel_result=$(cat "${iperf_file}")
+	parallel_result=$(cat "${iperf_file}")
+	export parallel_result=$(printf "%0.3f\n" $parallel_result)
 	export parallel_units="bits per second"
 
-	if [ "$COLLECT_ALL" == "true" ]; then
-		iperf3_all_collect_results
-	else
-		metrics_json_init
-		metrics_json_start_array
-
-		local json="$(cat << EOF
-		{
-			"parallel": {
-				"Result" : $parallel_result,
-				"Units" : "$parallel_units"
-			}
+	[ -z "${parallel_result}" ] && die "Iperf3 was unable to collect Parallel workload results."
+	[ "$COLLECT_ALL" == "true" ] && return
+
+	metrics_json_init
+	metrics_json_start_array
+
+	local json="$(cat << EOF
+	{
+		"parallel": {
+			"Result" : "${parallel_result}",
+			"Units" : "${parallel_units}"
 		}
+	}
 EOF
 )"
-		metrics_json_add_array_element "$json"
-		metrics_json_end_array "Results"
-	fi
+	metrics_json_add_array_element "${json}"
+	metrics_json_end_array "Results"
 }
 
 function iperf3_cpu() {
 	# Start server
```

Comment:
is this comment still relevant in this part of the function? iiuc, the iperf server is started in the `iperf3_start_deployment` function. I see this same comment in other functions.

Reply:
we can't get rid of this comment, thanks @chavafg